### PR TITLE
Fix some sonar issues

### PR DESCRIPTION
**Pull request by MaratGumarov**
- [ ] **Breaking change?** (if so, please describe the impact and migration path for existing application instances)

**What changes did you make?** (Give an overview)

**Is there anything you'd like reviewers to focus on?**

**How Has This Been Tested?** (put an "x" (case-sensitive!) next to an item)

- [ ] No need to
- [ ] Manually (please, describe, if necessary)
- [ ] Unit checks
- [ ] Integration checks
- [x] Covered by existing automation

**Checklist** (put an "x" (case-sensitive!) next to all the items, otherwise the build will fail)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (e.g. **ENVIRONMENT VARIABLES**)
- [x] My changes generate no new warnings (e.g. Sonar is happy)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged

Check out [Contributing](https://github.com/provectus/kafka-ui/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/provectus/kafka-ui/blob/master/CODE-OF-CONDUCT.md)

**A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/tests/ConnectorsTests.java [hunk 1]
```diff
@@ -71,7 +74,7 @@ void updateConnector() {
     @SneakyThrows
     @DisplayName("should delete connector")
     @Test
-    @Disabled
+    @Disabled("disable test due 500 error during create connector via api")
```
Comment (Haarolean) [inline kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/tests/ConnectorsTests.java:77]: we can revert these, these tests are being fixed within #1314 #1480

File: kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/base/BaseTest.java [hunk 2]
```diff
@@ -24,6 +24,7 @@
 
 @Slf4j
 @DisplayNameGeneration(CamelCaseToSpacedDisplayNameGenerator.class)
+@SuppressWarnings("squid:S2187")
```
Comment (Haarolean) [inline kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/base/BaseTest.java:27]: ?

File: kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/SmokeTests.java [hunk 3]
```diff
@@ -3,12 +3,12 @@
 import com.provectus.kafka.ui.base.BaseTest;
 import io.qameta.allure.Issue;
 import lombok.SneakyThrows;
-import org.junit.Ignore;
 import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
 
 @Disabled // TODO #1480
+@SuppressWarnings("squid:S2699")
```
Comment (Haarolean) [inline kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/SmokeTests.java:11]: ?

File: kafka-ui-api/src/test/java/com/provectus/kafka/ui/util/jsonschema/AvroJsonSchemaConverterTest.java [hunk 4]
```diff
@@ -14,9 +14,9 @@
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
 
-public class AvroJsonSchemaConverterTest {
+class AvroJsonSchemaConverterTest {
   @Test
-  public void avroConvertTest() throws URISyntaxException, JsonProcessingException {
+  void avroConvertTest() throws URISyntaxException, JsonProcessingException {
```
Comment (Haarolean) [inline kafka-ui-api/src/test/java/com/provectus/kafka/ui/util/jsonschema/AvroJsonSchemaConverterTest.java:19]: As far as I remember, junit tests should be public. If that's the case, let's make sonar ignore this stuff for test classes.

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/util/jsonschema/AvroJsonSchemaConverter.java [hunk 5]
```diff
@@ -55,12 +56,12 @@ private FieldSchema convertSchema(String name, Schema schema,
...
             return new MapFieldSchema(convertSchema(name, schema.getValueType(), definitions, ref));
           } else {
-            return createObjectSchema(name, schema, definitions, ref);
+            return createObjectSchema(schema, definitions, ref);
           }
         case ARRAY:
           return createArraySchema(name, schema, definitions);
         default:
-          throw new RuntimeException("Unknown type");
+          throw new KafkaUiRuntimeException("Unknown json type");
```
Comment (Haarolean) [inline kafka-ui-api/src/main/java/com/provectus/kafka/ui/util/jsonschema/AvroJsonSchemaConverter.java:64]: that looks like fixing a code smell just for the sake of getting rid of a warning. Please invent some better named exception isntead :)

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/strategy/ksql/statement/BaseStrategy.java [hunk 6]
```diff
@@ -16,15 +16,15 @@
...
+  protected static final String KSQL_REQUEST_PATH = "ksql";
+  protected static final String QUERY_REQUEST_PATH = "query";
   private static final String MAPPING_EXCEPTION_ERROR = "KSQL DB response mapping error";
   protected String host = null;
   protected KsqlCommandDTO ksqlCommand = null;
 
   public String getUri() {
     if (this.host != null) {
-      return this.host + this.getRequestPath();
+      return this.host + "/" + this.getRequestPath();
```
Comment (Haarolean) [inline kafka-ui-api/src/main/java/com/provectus/kafka/ui/strategy/ksql/statement/BaseStrategy.java:27]: Is that a thing sonar is whining about? Such a weird case tbh

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/ksql/response/ResponseParser.java [hunk 7]
```diff
@@ -13,6 +13,7 @@
 import java.util.stream.Collectors;
 import org.springframework.web.reactive.function.client.WebClientResponseException;
 
+@SuppressWarnings("squid:S1192")
```
Comment (Haarolean) [inline kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/ksql/response/ResponseParser.java:16]: ?

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/MessagesService.java [hunk 8]
```diff
@@ -48,6 +48,7 @@
 @Service
 @RequiredArgsConstructor
 @Slf4j
+@SuppressWarnings("squid:S2589")
```
Comment (Haarolean) [inline kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/MessagesService.java:51]: ?

**Review by Haarolean — CHANGES_REQUESTED** [review]

File: kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/tests/ConnectorsTests.java (hunk 1 above)
Comment (MaratGumarov) [reply kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/tests/ConnectorsTests.java:77]: Ok

File: kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/SmokeTests.java (hunk 3 above)
Comment (MaratGumarov) [reply kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/SmokeTests.java:11]: Sonar believes that a class with a name containing "Test" must contain tests. I'll see what I can do to make it look more elegant

File: kafka-ui-api/src/test/java/com/provectus/kafka/ui/util/jsonschema/AvroJsonSchemaConverterTest.java (hunk 4 above)
Comment (MaratGumarov) [reply kafka-ui-api/src/test/java/com/provectus/kafka/ui/util/jsonschema/AvroJsonSchemaConverterTest.java:19]: regarding java:S5786, JUnit5 test classes and methods should have default package visibility ([docs](https://junit.org/junit5/docs/current/user-guide/#writing-tests-classes-and-methods))

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/util/jsonschema/AvroJsonSchemaConverter.java (hunk 5 above)
Comment (MaratGumarov) [reply kafka-ui-api/src/main/java/com/provectus/kafka/ui/util/jsonschema/AvroJsonSchemaConverter.java:64]: Ok)

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/strategy/ksql/statement/BaseStrategy.java (hunk 6 above)
Comment (MaratGumarov) [reply kafka-ui-api/src/main/java/com/provectus/kafka/ui/strategy/ksql/statement/BaseStrategy.java:27]: Yes, it,s java:S1075 (URIs should not be hardcoded). The problem was in the `KSQL_REQUEST_PATH = "/ksql"; QUERY_REQUEST_PATH = "/query"` constants that stored the "/". Sonar does not recommend doing this

File: kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/base/BaseTest.java (hunk 2 above)
Comment (MaratGumarov) [reply kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/base/BaseTest.java:27]: Sonar believes that a class with a name containing "Test" must contain tests. I'll see what I can do to make it look more elegant

File: kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/SmokeTests.java (hunk 3 above)
Comment (MaratGumarov) [reply kafka-ui-e2e-checks/src/test/java/com/provectus/kafka/ui/SmokeTests.java:11]: Removed @SuppressWarnings. Seems like we need to fix these tests first

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/MessagesService.java (hunk 8 above)
Comment (MaratGumarov) [reply kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/MessagesService.java:51]: The problem is the generated CreateTopicMessageDTO, which can actually return null.  We have tests that check this

**Review by Haarolean — APPROVED** [review]

**Comment by Haarolean** [comment]
do not merge before #2125

**Pull request closed without merging**